header.go

```diff
@@ -270,16 +269,162 @@ func fixMangledMediaType(mtype, sep string) string {
 	return mtype
 }
 
-// fixUnquotedSpecials as defined in https://www.w3.org/Protocols/rfc1341/4_Content-Type.html
+func consumeParam(s string) (consumed, rest string) {
```

Comment:
I think a brief comment (with a sample input/output) for this function would help people picture what it does.